### ISSUE TITLE
Multiyear span

### ISSUE THREAD

**Issue by Olghn** [micz/ThunderStats#389]
It seems custom date ranges are limited to a calendar year.

I would quite like to see total emails per year back to the beginning of email. Also perhaps per month.
A use case I have for this is consolidating Thunderbird email folders that my (late) father had accumulated over many years on multiple machines.

One thing it lets me do is create a little chart/timeline of activity for each machine or account.

Another is it lets me verify that I correctly migrated the email to the machine I am consolidating the mail on - if I see one machine had emails mainly from 2017-2021, then after migration I should see a gap in that date range filled in the consolidated mailbox.

**Comment by micz** [comment]
I tested ThunderStats with 731 days and I found a bug in the days view, that is blank.
I filed issue #390.

This is the screenshot:
![photo_2024-11-19_21-09-31](https://github.com/user-attachments/assets/8cf49a26-f5a0-4c2f-909f-91e4ba38204f)


Clicking on the other views (weeks, months and years), it works, as shown here (it's the weeks view of the same query):
![photo_2024-11-19_21-11-23](https://github.com/user-attachments/assets/34b90d1b-1b08-4d35-b360-ea3fd44657e7)


The job took more than 2 minutes, but it eventually finished.

Is this the behavior you're experiencing?
Have you tried changing the view to weeks, months, or years?
Have you found any errors in the logs?
Thank you.
